packages/shared/src/schemas.ts

```diff
@@ -10,12 +10,34 @@ export const quoteQuerySchema = z.object({
 
 export type QuoteQueryParams = z.infer<typeof quoteQuerySchema>;
 
+export type ByteString = string | `0x${string}`;
+export interface CcmMetadata {
+  gas_budget: string | number;
+  message: ByteString;
+  source_address: string;
+  source_chain: 'Bitcoin' | 'Ethereum' | 'Polkadot';
+  cf_parameters?: ByteString;
+}
+
+export const ccmMetadataSchema = z.object({
+  gas_budget: z.union([numericString, z.number()]),
+  message: z.union([hexString, z.string()]),
+  source_address: z.string(),
```

Comment:
what kind of address is this?

Reply:
if it is the token address, we can infer it from the `srcAsset`

Reply:
@acdibble i'm not sure. how do i find out?